src/kirby/components/calendar/calendar.component.ts

```diff
@@ -0,0 +1,333 @@
+      this.disabledDates.forEach((disabledDate) => {
+        if (moment(disabledDate).isSame(date, 'day')) {
+          isDisabled = true;
+        }
+      });
+    }
+    return isDisabled;
+  }
+
+  refreshActiveMonth() {
+    // IMPORTANT: Moment startOf|endOf functions mutates the date!
+    // Clone the date before mutating:
+    const monthStart = this.activeMonth.clone().startOf('month');
+    const monthEnd = this.activeMonth.clone().endOf('month');
+    const startOfFirstWeek = monthStart.clone().startOf('isoWeek');
+    const endOfLastWeek = monthEnd.clone().endOf('isoWeek');
+    const totalDayCount = endOfLastWeek.diff(startOfFirstWeek, 'days') + 1;
+    const today = moment().startOf('day');
+    const daysArray = Array.from(Array(totalDayCount).keys());
+
+    const days: CalendarCell[] = daysArray.map((number) => {
+      const momentDate = startOfFirstWeek.clone().add(number, 'days');
+      const day = this.getCalendarDay(momentDate, today, monthStart);
+
+      const isSelectable = this.isSelectable(day, momentDate);
+      const isSelected = this.isSameDay(this.selectedDate, momentDate);
+      const cell = {
+        date: momentDate.date(),
+        isCurrentMonth: day.isCurrentMonth,
+        isSelectable,
+        isSelected,
+        cssClasses: this.getCssClasses(day, isSelectable, isSelected),
+      };
+      if (isSelected) {
+        this.selectedDay = cell;
+      }
+      return cell;
+    });
+    this.month = this.chunk(days, 7);
+  }
+
+  private getCalendarDay(
+    date: moment.Moment,
+    today: moment.Moment,
+    monthStart: moment.Moment
+  ): CalendarDay {
+    return {
+      isToday: date.isSame(today, 'day'),
+      isPast: date.isBefore(today),
+      isFuture: date.isAfter(today),
+      isWeekend: date.isoWeekday() === 6 || date.isoWeekday() === 7,
+      isCurrentMonth: date.isSame(monthStart, 'month'),
+      isDisabled: this.isDisabledDate(date),
+    };
+  }
+
+  private isSelectable(day: CalendarDay, date: moment.Moment) {
+    return (
+      !day.isDisabled &&
+      day.isCurrentMonth &&
+      !(this.disableWeekends && day.isWeekend) &&
+      !(this.disablePastDates && day.isPast) &&
+      !(this.disableFutureDates && day.isFuture) &&
+      !(this.minDate && date.isBefore(this.minDate, 'day')) &&
+      !(this.maxDate && date.isAfter(this.maxDate, 'day'))
+    );
+  }
+
+  private getCssClasses(day: CalendarDay, isSelectable: boolean, isSelected: boolean) {
+    const cssClasses = {
+      'current-month': day.isCurrentMonth,
+      weekend: day.isWeekend,
+      today: day.isToday,
+      selectable: isSelectable,
+      selected: isSelected,
+      past: day.isPast,
+      disabled: day.isDisabled,
+    };
+    let cssClassString = 'day';
+    for (var key in cssClasses) {
+      if (cssClasses[key]) {
+        cssClassString += ' ' + key;
+      }
+    }
+    return cssClassString;
+  }
+
+  private chunk(array: any[], size: number) {
+    var results = [];
```

Comment:
Consider using lexically scoped `const`-keyword instead of "hoisted" `var`-keyword

Reply:
Fixed